www/js/tripconfirm/trip-confirm-services.js

```diff
@@ -27,24 +27,45 @@ angular.module('emission.tripconfirm.services', ['ionic', "emission.plugin.logge
             });
     }
     
+    var loadAndPopulateOptionsLocales = function (filename) {
```

Comment:
Instead of adding this new function, can't you just expand `loadAndPopulateOptions` to have the following logic.

- if lang is not 'en', try to read language specific json
- if that fails, fall back to english json
- if that also fails, fall back to english json sample

Reply:
As you suggested, I changed loadAndPopulate: https://github.com/e-mission/e-mission-phone/pull/584/commits/5b947033d941f187ae6757678ccc37d92bafa1a7.

I used `lang` instead of `filename` as parameters since we were repeating it twice (for Modes and Purposes).